src/models/Scholarships.ts

```diff
@@ -72,10 +71,20 @@ class Scholarships extends FirestoreCollection<ScholarshipData> {
 
     query = query.limit(5).startAfter(lastDoc || 0);
 
-    const snapShot = await query.get();
-    const last = snapShot.docs[snapShot.docs.length - 1];
-    setLastDoc(last);
-    return FirestoreCollection.list(query);
+    // const snapShot = await query.get();
+    // const last = snapShot.docs[snapShot.docs.length - 1];
+    // setLastDoc(last);
+    return FirestoreCollection.list(query).then((results) => {
+      let lastDocument: any;
+      query.get().then((snapshot) => {
+        lastDocument = snapshot.docs[snapshot.docs.length - 1];
+      });
+      const next = () => {
+        return this.list(opts, lastDocument);
+      };
+
+      return { results, next };
+    });
   }
```

Comment:
This is a handler for what to do after listing completes and you have the results. You shouldn't need to do anything with `query.get()` because `results` already contains the results. I think what you want here is this:

```js
    return FirestoreCollection.list(query).then((results) => {
      return {results, next: results[results.length - 1].id};
    });
  }
```
or more succinctly:
```js
    return FirestoreCollection.list(query).then((res) => ({res, next: res[res.length - 1].id}));
  }
```

Reply:
Though, looking at the docs for [startAfter](https://firebase.google.com/docs/reference/js/firebase.firestore.Query#startafter) it looks like this is something that should probably be done in FirestoreCollection instead as the results from `FirestoreCollection.list(query)` are already [transformed](https://github.com/beyondhb1079/s4us/blob/main/src/models/base/FirestoreCollection.ts#L29) from `DocumentSnapshot`.